crates/iota-indexer/src/apis/governance_api.rs

```diff
@@ -561,6 +563,50 @@ impl GovernanceReadApi {
     }
 }
 
+/// Backfill missing rates for some epochs due to safe mode. If a rate is
+/// missing for epoch e, we will use the rate for epoch e-1 to fill it. Rates
+/// returned are in descending order by epoch.
+fn backfill_rates(
+    mut rates: Vec<(EpochId, PoolTokenExchangeRate)>,
+) -> Vec<(EpochId, PoolTokenExchangeRate)> {
+    if rates.is_empty() {
+        return rates;
+    }
+    // ensure epochs are processed in increasing order
+    rates.sort_unstable_by_key(|(epoch_id, _)| *epoch_id);
+
+    // Check if there are any gaps in the epochs
+    let min_epoch = rates.first().unwrap().0;
+    let max_epoch = rates.last().unwrap().0;
```

Comment:
Nit: It is typically preferred to use [expect as precondition](https://doc.rust-lang.org/std/error/index.html#common-message-styles) instead of unwrapping, and use pattern matching instead of indexing the tuple.
```suggestion
    let (min_epoch, _) = rates.first().expect("rates should not be empty");
    let (max_epoch, _) = rates.last().expect("rates should not be empty");
```

Reply:
Got it. Thanks.